### PR TITLE
9788 viewer in eumapi

### PR DESCRIPTION
**Pull request by gfiorav**
Please CR @javitonino

### CODE REVIEW

File: app/controllers/carto/api/organization_users_controller.rb [hunk 1]
```diff
@@ -130,16 +133,26 @@ def destroy
...
-               :soft_obs_snapshot_limit, :soft_obs_general_limit, :soft_twitter_datasource_limit)
+        @create_params ||=
+          permit(COMMON_MUTABLE_ATTRIBUTES + [:username, :viewer])
       end
 
       # TODO: Use native strong params when in Rails 4+
       def update_params
-        permit(:email, :password, :quota_in_bytes, :soft_geocoding_limit, :soft_here_isolines_limit,
-               :soft_obs_snapshot_limit, :soft_obs_general_limit, :soft_twitter_datasource_limit)
+        @update_params ||= permit(COMMON_MUTABLE_ATTRIBUTES)
```
Comment (javitonino) [inline app/controllers/carto/api/organization_users_controller.rb]: Why can't you update the viewer field?

File: app/helpers/organization_users_helper.rb [hunk 2]
```diff
@@ -39,7 +39,7 @@ def current_viewer_is_owner?
   def permit(*permitted)
     hardened_params = params.dup
 
-    hardened_params.keep_if { |k, _v| permitted.include?(k.to_sym) }
+    hardened_params.keep_if { |k, _v| permitted.flatten.include?(k.to_sym) }
```
Comment (javitonino) [inline app/helpers/organization_users_helper.rb]: If this is not used elsewhere, you could remove `flatten` here and remove the `*` from the method signature. Not important though.

File: app/models/carto/user_creation.rb [hunk 3]
```diff
@@ -210,6 +211,7 @@ def initialize_user
     @cartodb_user.soft_obs_snapshot_limit = soft_obs_snapshot_limit unless soft_obs_snapshot_limit.nil?
     @cartodb_user.soft_obs_general_limit = soft_obs_general_limit unless soft_obs_general_limit.nil?
     @cartodb_user.soft_twitter_datasource_limit = soft_twitter_datasource_limit unless soft_twitter_datasource_limit.nil?
+    @cartodb_user.viewer = viewer if viewer
```
Comment (javitonino) [inline app/models/carto/user_creation.rb]: `if viewer` is unnecessary as the `viewer` field is not-nullable, right?

File: app/controllers/carto/api/organization_users_controller.rb (hunk 1 above)
Comment (gfiorav) [inline app/controllers/carto/api/organization_users_controller.rb]: It's a mess apparently. Not as simple as updating the value :/

File: app/helpers/organization_users_helper.rb (hunk 2 above)
Comment (gfiorav) [inline app/helpers/organization_users_helper.rb]: Since it preserves the methods functionality and the abstractions make sense, I'd leave it!

File: app/models/carto/user_creation.rb (hunk 3 above)
Comment (gfiorav) [inline app/models/carto/user_creation.rb]: I'm not sure that if you expressly set something to null, the default thingies work as expected. Like: In a method with optional params, if you set one to nil, default values don't apply.

Comment (javitonino) [inline app/models/carto/user_creation.rb]: Not what I meant. Let me try again. `viewer` here is reading the field from the database (user_creations table) and that field cannot be null, so I don't think `viewer` can return `nil` here under any circumstances.

I think the `if  viewer` will always be true and thus is not needed. But anyway, not important, so feel free to ignore it.
